src/Mono.Android/Xamarin.Android.Net/AndroidClientHandler.cs

```diff
@@ -272,10 +272,12 @@ string EncodeUrl (Uri url)
 				if(Logger.LogNet)
 					Logger.Log (LogLevel.Info, LOG_APP, " cancelled");
 
-				httpConnection.Disconnect ();
+				await Task.Run (() => httpConnection.Disconnect ()).ConfigureAwait (false);
 				cancellationToken.ThrowIfCancellationRequested ();
 			}
-			cancellationToken.Register (httpConnection.Disconnect);
+			cancellationToken.Register (async () => {
+				await Task.Run (() => httpConnection.Disconnect ());
```

Comment:
I don't know if this call is correct. The [`CancellationToken.Register()`](https://msdn.microsoft.com/en-us/library/dd321635(v=vs.110).aspx) docs state:

> The current [ExecutionContext](https://msdn.microsoft.com/en-us/library/system.threading.executioncontext(v=vs.110).aspx), if one exists, will be captured along with the delegate and will be used when executing it.

Considering that the *above* `httpClient.Disconnect()` is turned into an `await Task.Run(...).ConfigureAwait(false)`, I have to wonder if *this* callback is *also* needs a `.ConfigureAwait(continueOnCapturedContext:false)`.